spark3/src/main/java/org/apache/iceberg/spark/SparkSessionCatalog.java

```diff
@@ -242,6 +242,13 @@ public void renameTable(Identifier from, Identifier to) throws NoSuchTableExcept
 
   @Override
   public final void initialize(String name, CaseInsensitiveStringMap options) {
+    if (options.containsKey("type") && options.get("type").equalsIgnoreCase("hive") && options.containsKey("uri")) {
+      throw new UnsupportedOperationException("Cannot set an alternative uri when" +
+              " using the SparkSessionCatalog, " +
+              "make an alternative SparkCatalog if the " +
+              "Spark Session catalog and Iceberg " +
+              "catalog should contact different metastores.");
+    }
```

Comment:
If this is needed, should we check that the configured `uri` isn't already equal to the value of the metastore URI configured via `spark.hadoop.hive.metastore.uris` or any of the other ways of setting it.

This would be a breaking change for people who have `uri` configured on the SparkSessionCatalog and have it correctly set (which I believe should be everybody as `uri` is currently required if I'm not mistaken).

Reply:
Agreed, this will make it backward compatible to some extent but allowing this will make the config set up more prone to error as when someone is changing the thrift server uri they would now have to do it at two places. Its more of a tradeoff between the two situations - backward compatible vs one config one place. @RussellSpitzer @rdblue what are your opinions?

Reply:
I agree that we should only throw if it doesn't match. We can also warn if it does.

Reply:
Do we need to update any docs on the Iceberg site for this?

Reply:
Made the change, pls check, also I was thinking we can add this check doc here - https://iceberg.apache.org/spark-configuration/#replacing-the-session-catalog, if it already not present elsewhere.